### PR TITLE
meta: rename project to `tracing`

### LINKED ISSUE #95: Rename to tracing
### Crates

all

### Motivation

As described in https://github.com/tokio-rs/tokio/issues/1159#issuecomment-503782726, the name `tokio-trace` has created the incorrect impression that using this library _requires_ the tokio runtime.

### Proposal

We are in the process of moving the main crates from the `tokio` repository to this repository, and have renamed the repository to `tracing`. At the same time, we will rename the nursery crates as well.

- [x] Rename repository
- [x] Move core crates (tokio-rs/tokio#1159)
- [x] Rename core crates
- [x] Rename nursery crates
- [x] Update README, CONTRIBUTING, and RustDocs
- [x] Update any links

### Alternatives

We may also want to rename some of the compatibility crates to use a different format --- I'm considering renaming `tokio-trace-tower` to `tower-tracing` instead of `tracing-tower`.

### PR DESCRIPTION
**Pull request by hawkw**
See #95 

This branch renames everything from `tokio-trace` to `tracing`.

This branch _also_ updates the minimum Rust version to 1.34.0,
to the shock and horror of the millions of `tracing` users still
on Rust 1.26.0; I didn't want to add all the 
`#![allow(bare_trait_objects)]` attributes necessary to placate
the nightly CI build.

Closes #95

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,33 +4,15 @@ rust:
...
-- cargo test --verbose -p tokio-trace-futures
-- cargo test --verbose -p tokio-trace-subscriber
-- cargo test --verbose -p tokio-trace-log
-- cargo test --verbose -p tokio-trace-env-logger
-- cargo test --verbose -p tokio-trace-macros
-- cargo test --verbose -p tokio-trace-serde
-- cargo test --verbose -p tokio-trace-fmt
-- cargo test --verbose -p tokio-trace-slog
-
+- cargo test --all
```
Comment (jonhoo) [inline .travis.yml]: Should this keep `--verbose`? Do we also need to take features into account here? Testing with `--no-default-features` for example can be handy. Personally I also like doing a first `check` pass [like this](https://github.com/jonhoo/inferno/blob/994b61ff67862c7797d8bb944a057c7cf3852fe4/.travis.yml#L19-L21) to eliminate bad commits early.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -14,15 +14,15 @@ Please fill in as much of the template below as you're able.
 ### Version
 
 <!--
-List the versions of all `tokio-trace` crates you are using. The easiest way to get
+List the versions of all `tracing` crates you are using. The easiest way to get
```
Comment (jonhoo) [inline .github/ISSUE_TEMPLATE/bug_report.md]: I'm just going to assume here that you've searched for `tokio-trace` and `tokio_trace` across all files :p

File: Cargo.toml [hunk 3]
```diff
@@ -1,79 +1,17 @@
...
-    "tokio-trace-tower",
-    "tokio-trace-tower-http",
-    "tokio-trace-log",
-    "tokio-trace-env-logger",
-    "tokio-trace-slog",
-    "tokio-trace-macros",
-    "tokio-trace-subscriber",
-    "tokio-trace-serde",
+    "tracing",
+    "tracing-core",
```
Comment (jonhoo) [inline Cargo.toml]: Ah, so this PR _also_ moves a bunch of stuff around then?

File: tracing-env-logger/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+[package]
```
Comment (jonhoo) [inline tracing-env-logger/Cargo.toml]: I wonder why these weren't picked up as file renames..

File: tracing-fmt/examples/yak_shave.rs [hunk 5]
```diff
@@ -1,34 +1,34 @@
...
-extern crate tokio_trace;
-extern crate tokio_trace_fmt;
+extern crate tracing;
+extern crate tracing_fmt;
 
-use tokio_trace::Level;
+use tracing::Level;
 
 fn shave(yak: usize) -> bool {
-    span!(Level::TRACE, "shave", yak = yak).enter(|| {
```
Comment (jonhoo) [inline tracing-fmt/examples/yak_shave.rs]: This pollutes the diff a bit, but is probably fine.

File: tracing-fmt/src/filter/reload.rs [hunk 6]
```diff
@@ -16,7 +16,7 @@ where
     _f: PhantomData<fn(N)>,
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug)]
```
Comment (jonhoo) [inline tracing-fmt/src/filter/reload.rs]: This change seems unrelated?

File: tracing-fmt/src/filter/reload.rs [hunk 7]
```diff
@@ -41,10 +41,8 @@ impl<F, N> Filter<N> for ReloadFilter<F, N>
 where
     F: Filter<N>,
 {
-    fn callsite_enabled(&self, _: &Metadata, _: &Context<N>) -> Interest {
-        // TODO(eliza): When tokio-rs/tokio#1039 lands, we can allow our
-        // interest to be cached. For now, we must always return `sometimes`.
-        Interest::sometimes()
+    fn callsite_enabled(&self, metadata: &Metadata, ctx: &Context<N>) -> Interest {
+        self.inner.read().callsite_enabled(metadata, ctx)
```
Comment (jonhoo) [inline tracing-fmt/src/filter/reload.rs]: This also seems unrelated?

File: tracing-fmt/src/filter/reload.rs [hunk 8]
```diff
@@ -92,10 +90,14 @@ where
...
-        let mut inner = inner.write();
-        f(&mut *inner);
-        // TODO(eliza): When tokio-rs/tokio#1039 lands, this is where we would
-        // invalidate the callsite cache.
+
+        let mut lock = inner.write();
+        f(&mut *lock);
+        // Release the lock before rebuilding the interest cache, as that
+        // function will lock the new filter.
+        drop(lock);
```
Comment (jonhoo) [inline tracing-fmt/src/filter/reload.rs]: And this seems unrelated. Subject to a different PR?

File: tracing-futures/Cargo.toml [hunk 9]
```diff
@@ -8,10 +8,10 @@ default = ["tokio"]
 
 [dependencies]
 futures = "0.1"
-tokio-trace = "0.1.0"
+tracing = { path = "../tracing" }
```
Comment (jonhoo) [inline tracing-futures/Cargo.toml]: Oh, this is _excellent_! Now depending on `tracing` from git along with these crates will finally work again!

File: tracing-futures/src/executor.rs [hunk 10]
```diff
@@ -4,12 +4,11 @@ use futures::{
 };
 use {Instrument, Instrumented, WithDispatch};
 
-#[cfg(all(feature = "tokio", not(feature = "tokio-executor")))]
```
Comment (jonhoo) [inline tracing-futures/src/executor.rs]: Is the `tokio-executor` feature one that we want to keep?

File: tracing-macros/src/lib.rs [hunk 11]
```diff
@@ -10,33 +10,33 @@ macro_rules! trace_dbg {
...
-        use tokio_trace::callsite::Callsite;
-        use tokio_trace::{
+        use tracing::callsite::Callsite;
+        use tracing::{
             callsite,
             field::{debug, Value},
             Event, Id, Subscriber,
         };
         let callsite = callsite! {
-            name: concat!("event:trace_dbg(", stringify!($ex), ")"),
```
Comment (jonhoo) [inline tracing-macros/src/lib.rs]: This change seems unrelated?

File: tracing-macros/src/lib.rs [hunk 12]
```diff
@@ -10,33 +10,33 @@ macro_rules! trace_dbg {
...
-            kind: tokio_trace::metadata::Kind::EVENT,
+            name: stringify!($ex),
+            kind: tracing::metadata::Kind::EVENT,
             target: $target,
             level: $level,
-            fields: $ex
+            fields: value,
```
Comment (jonhoo) [inline tracing-macros/src/lib.rs]: This change also seems unrelated?

File: tracing-serde/examples/serde_shaved_yak.rs [hunk 13]
```diff
@@ -85,32 +85,32 @@ impl Subscriber for JsonSubscriber {
 }
 
 fn shave(yak: usize) -> bool {
-    span!(Level::TRACE, "shave", yak = yak).enter(|| {
```
Comment (jonhoo) [inline tracing-serde/examples/serde_shaved_yak.rs]: Also makes the diff noisy, but probably fine.

File: tracing-tower-http/examples/tower-h2-server.rs [hunk 14]
```diff
@@ -40,11 +40,7 @@ impl Body for RspBody {
     type Data = <Bytes as IntoBuf>::Buf;
     type Error = h2::Error;
 
-    fn is_end_stream(&self) -> bool {
```
Comment (jonhoo) [inline tracing-tower-http/examples/tower-h2-server.rs]: Why was this removed?

File: tracing-tower-http/examples/tower-h2-server.rs [hunk 15]
```diff
@@ -125,54 +121,49 @@ fn main() {
...
-                            .and_then(|_| {
-                                debug!("response finished");
-                                future::ok(())
-                            })
-                            .instrument(conn_span2);
-                        reactor.spawn(Box::new(serve));
-
-                        Ok((h2, reactor))
+            tracing_tower_http::InstrumentedMakeService::with_span(NewSvc, serve_span.clone());
+        let h2 = Server::new(new_svc, Default::default(), reactor.clone());
```
Comment (jonhoo) [inline tracing-tower-http/examples/tower-h2-server.rs]: This no longer happens under the span whereas it did before.

File: tracing-tower-http/examples/tower-h2-server.rs [hunk 16]
```diff
@@ -125,54 +121,49 @@ fn main() {
...
+                    .instrument(conn_span.clone());
+                reactor.spawn(Box::new(serve));
+
+                Ok((h2, reactor))
+            })
+            .map_err(|e| {
+                error!("serve error {:?}", e);
+            })
+            .map(|_| {})
+            .instrument(serve_span);
```
Comment (jonhoo) [inline tracing-tower-http/examples/tower-h2-server.rs]: Big fan of this change.

File: tracing-tower-http/src/lib.rs [hunk 17]
```diff
@@ -124,18 +124,17 @@ where
...
         } else {
             self.inner.poll_ready()
         }
     }
 
     fn call(&mut self, request: http::Request<B>) -> Self::Future {
-        let inner = &mut self.inner;
         let span = trace_span!(
-            parent: self.span.as_ref(),
+            parent: self.span.as_ref().and_then(Span::id),
```
Comment (jonhoo) [inline tracing-tower-http/src/lib.rs]: This change seems unrelated?

File: tracing-tower/src/lib.rs [hunk 18]
```diff
@@ -1,22 +1,23 @@
...
 extern crate futures;
-extern crate tokio_trace_futures;
+extern crate tracing_futures;
 
 use std::fmt;
-use tokio_trace::{field, Level};
-use tokio_trace_futures::{Instrument, Instrumented};
 use tower_service::Service;
 
+use tracing::{field, Level};
```
Comment (jonhoo) [inline tracing-tower/src/lib.rs]: Any particular reason you moved these away from the other `use` statements?

File: tracing-tower/src/lib.rs [hunk 19]
```diff
@@ -31,19 +32,15 @@ where
...
-        let inner = &mut self.inner;
-        span.enter(|| {
-            // TODO: custom `Value` impls for `http` types would be nice...
-            let span = span!(Level::TRACE, "request", request = &field::debug(&req));
-            let span2 = span.clone();
-            span.enter(move || inner.call(req).instrument(span2))
-        })
+        // TODO: custom `Value` impls for `http` types would be nice...
+        let span =
+            span!(Level::TRACE, parent: &self.span, "request", request = &field::debug(&req));
```
Comment (jonhoo) [inline tracing-tower/src/lib.rs]: Is it necessary to take the `&` manually here?

File: tracing/README.md [hunk 20]
```diff
@@ -0,0 +1,192 @@
+# tracing
```
Comment (jonhoo) [inline tracing/README.md]: I'm going to assume that this has basically stayed the same.

File: tracing/tests/macros.rs [hunk 21]
```diff
@@ -533,7 +533,7 @@ fn error_root() {
 
 #[test]
 fn event_with_parent() {
-    let p = span!(Level::TRACE, "im_a_parent!");
+    let _p = span!(Level::TRACE, "im_a_parent!");
```
Comment (jonhoo) [inline tracing/tests/macros.rs]: This shouldn't be necessary any more. You probably added this for the warning that was emitted due to the `parent:` bug. May also need to be undone elsewhere.

Comment (hawkw) [reply tracing/tests/macros.rs]: rustfix did this, will make sure the parent bug is actually resolved

File: tracing-tower-http/examples/tower-h2-server.rs (hunk 14 above)
Comment (hawkw) [reply tracing-tower-http/examples/tower-h2-server.rs]: the trait changed upstream; the old code no longer complied and I wanted a green CI build.

File: tracing-futures/Cargo.toml (hunk 9 above)
Comment (hawkw) [reply tracing-futures/Cargo.toml]: yup, however, this is also why there are apparently unrelated changes in this PR --- some of the nursery crates didn't compile with the master version of the core crates (which have breaking changes), so I merged changes in from a branch that updated them to work with the latest.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (hawkw) [reply .github/ISSUE_TEMPLATE/bug_report.md]: pretty much

**Comment by hawkw** [comment]
@jonhoo as I mentioned, most of the changes that seem unrelated were necessary to get the nursery crates to compile after changing to a path dependency on `tracing`/`tracing-core`. It might be better to back out those changes and leave the nursery crates depending on `tokio-trace`/`tokio-trace-core` from crates.io. We can then update dependencies to `tracing` in a subsequent branch. If you think that's preferable, I'd be happy to make that change.

**Comment by jonhoo** [comment]
I _think_ I would prefer that as it would make the change a little easier to follow. I trust you to have correctly ported over the changes though, so it may be that I'm just making a fuss about nothing :)

**Comment by hawkw** [comment]
@jonhoo PR #99 is a version of this change with the nursery crate path dep updates backed out.
